python/cudf_polars/cudf_polars/experimental/base.py

```diff
@@ -248,16 +250,61 @@ def new_parent(
         )
 
 
+class JoinKey:
+    """
+    Join-key information.
+
+    Parameters
+    ----------
+    column_stats
+        Column statistics for the join key.
+
+    Notes
+    -----
+    This class is used to track join-key information.
+    It is used to track the columns being joined on
+    and the estimated unique-value count for the join key.
+    """
+
+    column_stats: tuple[ColumnStats, ...]
+    unique_count_estimate: int | None
+    """Estimated unique-value count from join heuristics."""
+
+    def __init__(self, *column_stats: ColumnStats) -> None:
+        self.column_stats = column_stats
+        self.unique_count_estimate = None
+
+    @cached_property
+    def source_row_count(self) -> int | None:
+        """Return the estimated row-count of the source columns."""
+        return min(
```

Comment:
When do we have multiple column stats with their own row counts? Joining on the same column name from multiple source tables?